packages/flutter/lib/src/material/bottom_navigation.dart

```diff
@@ -0,0 +1,536 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:collection' show Queue;
+
+import 'package:flutter/widgets.dart';
+import 'package:vector_math/vector_math_64.dart' show Vector3;
+
+import 'colors.dart';
+import 'icon.dart';
+import 'icon_theme.dart';
+import 'icon_theme_data.dart';
+import 'ink_well.dart';
+import 'material.dart';
+import 'theme.dart';
+
+const double kBottomNavigationHeight = 60.0;
+const double _kActiveMaxWidth = 168.0;
+const double _kInactiveMaxWidth = 96.0;
+
+/// Closure used in [BottomNavigation] to handle action taps. The [animation]
+/// argument is useful in order to coordinate with the action's animation. The
+/// [index] argument specifies which action has been tapped.
+typedef void BottomNavigationCallback(Animation<double> animation, int index);
+
+/// The [BottomNavigation] type which affects its layout.
```

Comment:
Avoid reiterating a type name as its definition. In this case it would be enough to say: `Defines the layout for a [BottomNavigationBar].`